### DISCUSSION TITLE
Theme customisations not working?

### DISCUSSION THREAD

**Discussion by founderpad** [chakra-ui/chakra-ui#4857] [Q&A]
What am I doing wrong here? I've checked the documentation.

Thanks all!

```
const customTheme = extendTheme({
	fonts: {
		heading: 'Roboto',
		body: 'Roboto',
		text: 'Roboto',
		menubar: 'Roboto',
		button: 'Roboto'
	},
	styles: {
		input: { // doesn't work???
			_hover: {
				backgroundColor: 'red'
			}
		}
	},
        // None of these are working?
	components: {
		Button: {
			baseStyle: {
				rounded: 'md',
				_focus: { boxShadow: 'none' },
                                 _hover: { backgroundColor: 'brand.500' }, 
				fontWeight: 'medium'
			}
		},
		Tabs: {
			baseStyle: {
				outline: 0,
				_focus: { boxShadow: 'none' },
				fontWeight: 'bold'
			}
		},
		Input: {
			baseStyle: {
				// rounded: 'md',

				// _hover: {
				// 	bgColor: 'red.500'
				// }
			}
		},
		MenuItem: {
			baseStyle: {
				bgColor: 'transparent'
			}
		},
		Select: {
			baseStyle: {
				// bgColor: 'red'
			}
		},
		Link: {
			// variants: {
			// 	primary: ({ colorScheme = 'fpGrey' }) => ({
			// 		color: `${colorScheme}.500`,
			// 		_hover: {
			// 			color: `${colorScheme}.600`
			// 		}
			// 	})
			// }
		},
		Badge: {
			sizes: {
				lg: {
					h: 12,
					minW: 12,
					fontSize: 'lg',
					px: 6
				},
				md: {
					h: 10,
					minW: 10,
					fontSize: 'md',
					px: 4
				},
				sm: {
					h: 8,
					minW: 8,
					fontSize: 'sm',
					px: 3
				},
				xs: {
					h: 6,
					minW: 6,
					fontSize: 'xs',
					px: 2
				}
			}
		}
	}
});
```

**Comment by primos63** [comment]
You're not correctly defining your theme extension for the components. You have to pay attention to the default theme source for the component. There's a button/link for that on most of the components.

You can look at my [CodeSandbox](https://codesandbox.io/s/chakra-ui-theme-extension-w5u2n) I created for questions from Github and Discord. If that doesn't help get you unstuck, let me know.

**Reply by lucasmaffazioli** [reply]
Thank you for the CodeSandbox, very useful!

But how can I discover on my own how can I find the correct keys I can use to extend components? 

For example, now I cant change the text color of the tab, something that I think should be simple for me to grasp:

![image](https://user-images.githubusercontent.com/12513049/139135795-8a395322-8fae-4011-a03b-4ca72d0ee24a.png)

![image](https://user-images.githubusercontent.com/12513049/139135842-de28704f-13d4-4b31-95df-c2f79ad65e18.png)

bg works, but color don't, how?

**Reply by primos63** [reply]
You always need to look at the theme source. Part of that CSB is a how-to on extending a theme.
![image](https://user-images.githubusercontent.com/2768849/139168007-26ad61f8-b7b1-4c8d-8f80-63402c60ef64.png)


For `Tabs` these are the settings for baseStyle.
```javascript
const baseStyle: PartsStyleFunction<typeof parts> = (props) => ({
  root: baseStyleRoot(props),
  tab: baseStyleTab(props),
  tablist: baseStyleTablist(props),
  tabpanel: baseStyleTabpanel,
})
```
Note that for `tab` the function `baseStyleTab(props)` is called.
```javascript
const baseStyleTab: SystemStyleFunction = (props) => {
  const { isFitted } = props

  return {
    flex: isFitted ? 1 : undefined,
    transitionProperty: "common",
    transitionDuration: "normal",
    _focus: {
      zIndex: 1,
      boxShadow: "outline",
    },
  }
}
```
Here you see there is no `color` property. As you found out specifying `tab: { _selected: { color: 'red' }}` doesn't work. Why?
Well `Tabs` have variants. The default variant is `line`
```javascript
const variants: Record<string, PartsStyleInterpolation<typeof parts>> = {
  line: variantLine,
  enclosed: variantEnclosed,
  "enclosed-colored": variantEnclosedColored,
  "soft-rounded": variantSoftRounded,
  "solid-rounded": variantSolidRounded,
  unstyled: variantUnstyled,
}

const variantLine: PartsStyleFunction<typeof parts> = (props) => {
  const { colorScheme: c, orientation } = props
  const isVertical = orientation === "vertical"
  const borderProp = orientation === "vertical" ? "borderStart" : "borderBottom"
  const marginProp = isVertical ? "marginStart" : "marginBottom"

  return {
    tablist: {
      [borderProp]: "2px solid",
      borderColor: "inherit",
    },
    tab: {
      [borderProp]: "2px solid",
      borderColor: "transparent",
      [marginProp]: "-2px",
      _selected: {
        color: mode(`${c}.600`, `${c}.300`)(props),             // <=== Overwrites the baseStyle
        borderColor: "currentColor",
      },
      _active: {
        bg: mode("gray.200", "whiteAlpha.300")(props),
      },
      _disabled: {
        opacity: 0.4,
        cursor: "not-allowed",
      },
    },
  }
}
```
You can see that the variant setting for `tab: { _selected: { color: ... }}`. This will overwrite what you specified in the your theme's baseStyle.

So you need to specify the color in the `variants` object of your theme. I modified my Tabs example to show what needs to be done.
```javascript
  components: {
    Tabs: {
      variants: {
        line: {
          tab: {
            _selected: {
              color: "red.500"
            }
          }
        }
      },
      baseStyle: {
        tab: {
          _focus: {
            boxShadow: "none"
          },
          _hover: {
            bg: "blue.200"
          },
          _selected: {
            _hover: {
              bg: "blue.200"
            }
          }
        }
      }
    }
  }
  ```

**Reply by founderpad** [reply]
You've just removed A LOT of base styles I was applying in my components!! Wow, thank you so so much. I can now bake them into the theme which reduces a lot. Thank you!!!!!

**Reply by lucasmaffazioli** [reply]
Thank you so much @primos63, I now payed deserved attention to all text of the theme pages on the chakra-ui documentation, and thanks to yor explanation I'm able to change the tabs!
